### PR TITLE
Add proxy endpoint for kubernetes dashboard

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
It adds a new endpoint that allows access to Kubernetes Dashboard installed for the user cluster. Access will be limited to owners and editors of the project and they will be granted admin privileges in the Kubernetes Dashboard. Later on, we will reuse RBAC for the user clusters feature to limit the privileges.

`/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/dashboard/proxy`

**Special notes for your reviewer**:
I have cleaned up some code and moved it so it can be reused by the dashboard proxy.

**Does this PR introduce a user-facing change?**:

```release-note
Add a new proxy endpoint to access Kubernetes Dashboard for the user cluster.
```

### CODE REVIEW

File: api/pkg/handler/v1/common/common.go [hunk 1]
```diff
@@ -97,3 +112,147 @@ func (d CredentialsData) GetGlobalSecretKeySelectorValue(configVar *providerconf
...
+	cfg *rest.Config,
+	namespace string,
+	labelSelector string,
+	containerPort int) (*portforward.PortForwarder, *bytes.Buffer, error) {
+	pod, err := GetReadyPod(coreClient.Pods(namespace), labelSelector)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	fmt.Println(pod)
```
Comment (alvaroaleman) [inline api/pkg/handler/v1/common/common.go]: Remainder?

File: api/pkg/handler/v1/kubernetes-dashboard/dashboard.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+				return nil, nil
+			}
+
+			proxyUrl := &url.URL{
+				Scheme: "http",
+				Host:   fmt.Sprintf("127.0.0.1:%d", port),
+			}
+
+			// Override strict CSP policy for proxy
+			w.Header().Set("Content-Security-Policy", "default-src 'self'; script-src 'self' 'unsafe-inline'; object-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self'; media-src 'self'; frame-ancestors 'self'; frame-src 'self'; connect-src 'self'; font-src 'self' data:")
```
Comment (alvaroaleman) [inline api/pkg/handler/v1/kubernetes-dashboard/dashboard.go:112]: Is this needed?

File: api/pkg/handler/v1/kubernetes-dashboard/dashboard.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+// It's responsible for adjusting proxy request, so we can properly access Kubernetes Dashboard
+type dashboardProxyDirector struct {
+	proxyUrl        *url.URL
+	token           string
+	originalRequest *http.Request
+}
+
+func (self *dashboardProxyDirector) director() func(*http.Request) {
```
Comment (alvaroaleman) [inline api/pkg/handler/v1/kubernetes-dashboard/dashboard.go]: The `self` is very obviously coming from someone with a JS background :D

File: api/pkg/handler/v1/kubernetes-dashboard/dashboard.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+		req.Header.Set("Authorization", self.getAuthorizationHeader())
+	}
+}
+
+func (self *dashboardProxyDirector) getAuthorizationHeader() string {
+	return fmt.Sprintf("Bearer %s", self.token)
+}
+
+// We need to get proper path to Dashboard API and strip the URL from the Kubermatic API request part.
+func (self *dashboardProxyDirector) getBasePath(path string) string {
```
Comment (alvaroaleman) [inline api/pkg/handler/v1/kubernetes-dashboard/dashboard.go]: Oh, thats a nice way of doing this without requireing adjustments

File: api/pkg/handler/v1/kubernetes-dashboard/dashboard.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+	for _, info := range kubeconfig.AuthInfos {
+		if len(info.Token) > 0 {
+			return info.Token, nil
+		}
+	}
+
+	return "", errors.New("could not find bearer token in kubeconfig file")
+}
+
+func forwardPort(log *zap.SugaredLogger, forwarder *portforward.PortForwarder) error {
```
Comment (alvaroaleman) [inline api/pkg/handler/v1/kubernetes-dashboard/dashboard.go]: This could also go onto the `common` package or not?

**Review by alvaroaleman — COMMENTED** [review]
Some nits, looks really good!

File: api/pkg/handler/v1/common/common.go (hunk 1 above)
Comment (floreks) [reply api/pkg/handler/v1/common/common.go]: Yep. Forgot to remove it.

File: api/pkg/handler/v1/kubernetes-dashboard/dashboard.go (hunk 2 above)
Comment (floreks) [reply api/pkg/handler/v1/kubernetes-dashboard/dashboard.go:112]: Similar to the openshift console, we need to override it. I have also slightly modified it at the end to allow embedded fonts (`font-src 'self' data:"`).

File: api/pkg/handler/v1/kubernetes-dashboard/dashboard.go (hunk 3 above)
Comment (floreks) [reply api/pkg/handler/v1/kubernetes-dashboard/dashboard.go]: Actually I think it was due to some tutorial about go structures/interfaces where someone was using it. I'm using `this` or `self` usually within structures as they are very similar to classes from C++/Java :slightly_smiling_face:.

**Comment by floreks** [comment]
Done. PTAL

**Comment by alvaroaleman** [comment]
Modulo the lint failures, lgtm.
/approve

File: api/pkg/handler/v1/kubernetes-dashboard/dashboard.go (hunk 3 above)
Comment (alvaroaleman) [reply api/pkg/handler/v1/kubernetes-dashboard/dashboard.go]: Linter doesn't like it:
```
pkg/handler/v1/kubernetes-dashboard/dashboard.go:136:1: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (golint)
```

**Comment by floreks** [comment]
Unfortunately `make lint` does not work for me locally even when I'm not using `gomod`. It fails with compilation error for all packages in the project, but there is no actual error shown why it cannot compile :scream:

**Comment by maciaszczykm** [comment]
> Unfortunately make lint does not work for me locally even when I'm not using gomod.

Same here :|

**Comment by alvaroaleman** [comment]
> Unfortunately make lint does not work for me locally even when I'm not using gomod.

Do you have the right version of `golangci-lint`? It must match what is here: https://github.com/kubermatic/infra/blob/2d78f910e6dc534e93bc496470e5c3998ab1ceca/prow/jobs/kubermatic/kubermatic-presubmits.yaml#L536

Unfortunately `golangci-lint` doesn't seem to offer any way to figure out its version

**Comment by floreks** [comment]
@alvaroaleman  It seems that I do have the correct version.

```bash
11:53 $ golangci-lint --version
golangci-lint has version 1.17.1 built from 4ba2155 on 2019-06-10T09:06:49Z
11:53 $ make lint
./hack/ci/ci-run-lint.sh
level=info msg="[config_reader] Config search paths: [./ /home/floreks/Projekty/go/src/github.com/kubermatic/kubermatic/api /home/floreks/Projekty/go/src/github.com/kubermatic/kubermatic /home/floreks/Projekty/go/src/github.com/kubermatic /home/floreks/Projekty/go/src/github.com /home/floreks/Projekty/go/src /home/floreks/Projekty/go /home/floreks/Projekty /home/floreks /home /]"
level=info msg="[config_reader] Used config file ../.golangci.yml"
level=info msg="[lintersdb] Active 18 linters: [bodyclose deadcode errcheck goconst gocritic gocyclo gofmt golint gosimple govet ineffassign interfacer misspell staticcheck structcheck unconvert unused varcheck]"
level=info msg="[lintersdb] Optimized sublinters [staticcheck gosimple unused] into metalinter megacheck"
level=info msg="[loader] Go packages loading at mode load types and syntax took 9.882580538s"
level=info msg="[loader] SSA repr building timing: packages building 11.776483ms, total 84.073423ms"
level=info msg="[loader] Packages that do not compile: [github.com/kubermatic/kubermatic/api/pkg/crd/kubermatic/v1 github.com/kubermatic/kubermatic/api/pkg/api/v1 github.com/kubermatic/kubermatic/api/pkg/log github.com/kubermatic/kubermatic/api/pkg/util/restmapper github.com/kubermatic/kubermatic/api/pkg/controller/util/predicate github.com/kubermatic/kubermatic/api/pkg/util/workerlabel github.com/kubermatic/kubermatic/api/pkg/resources/certificates/triple github.com/kubermatic/kubermatic/api/pkg/cluster/client github.com/kubermatic/kubermatic/api/pkg/crd/client/clientset/versioned/scheme github.com/kubermatic/kubermatic/api/pkg/crd/client/clientset/versioned/typed/kubermatic/v1 github.com/kubermatic/kubermatic/api/pkg/crd/client/clientset/versioned github.com/kubermatic/kubermatic/api/pkg/crd/client/informers/externalversions/internalinterfaces github.com/kubermatic/kubermatic/api/pkg/crd/client/listers/kubermatic/v1 github.com/kubermatic/kubermatic/api/pkg/crd/client/informers/externalversions/kubermatic/v1 github.com/kubermatic/kubermatic/api/pkg/crd/client/informers/externalversions/kubermatic github.com/kubermatic/kubermatic/api/pkg/crd/client/informers/externalversions github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/models github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/addon github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/aws github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/azure github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/credentials github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/datacenter github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/digitalocean github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/gcp github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/hetzner github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/metric github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/openstack github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/operations github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/packet github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/project github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/serviceaccounts github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/tokens github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/users github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/versions github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client/vsphere github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/apiclient/client github.com/kubermatic/kubermatic/api/cmd/conformance-tests github.com/kubermatic/kubermatic/api/cmd/http-prober github.com/kubermatic/kubermatic/api/pkg/addon github.com/kubermatic/kubermatic/api/pkg/resources/certificates github.com/kubermatic/kubermatic/api/pkg/provider/cloud/azure github.com/kubermatic/kubermatic/api/pkg/provider/cloud/bringyourown github.com/kubermatic/kubermatic/api/pkg/provider/cloud/digitalocean github.com/kubermatic/kubermatic/api/pkg/provider/cloud/fake github.com/kubermatic/kubermatic/api/pkg/provider/cloud/gcp github.com/kubermatic/kubermatic/api/pkg/provider/cloud/hetzner github.com/kubermatic/kubermatic/api/pkg/provider/cloud/kubevirt github.com/kubermatic/kubermatic/api/pkg/provider/cloud/openstack github.com/kubermatic/kubermatic/api/pkg/provider/cloud/packet github.com/kubermatic/kubermatic/api/pkg/provider/cloud github.com/kubermatic/kubermatic/api/pkg/controller/cloud github.com/kubermatic/kubermatic/api/pkg/uuid github.com/kubermatic/kubermatic/api/pkg/provider/kubernetes github.com/kubermatic/kubermatic/api/pkg/resources/etcd/etcdrunning github.com/kubermatic/kubermatic/api/pkg/resources/nodeportproxy github.com/kubermatic/kubermatic/api/pkg/resources/vpnsidecar github.com/kubermatic/kubermatic/api/pkg/resources/cloudconfig github.com/kubermatic/kubermatic/api/pkg/resources/clusterautoscaler github.com/kubermatic/kubermatic/api/pkg/resources/controllermanager github.com/kubermatic/kubermatic/api/pkg/resources/dns github.com/kubermatic/kubermatic/api/pkg/resources/kubernetes-dashboard github.com/kubermatic/kubermatic/api/pkg/resources/machinecontroller github.com/kubermatic/kubermatic/api/pkg/resources/certificates/servingcerthelper github.com/kubermatic/kubermatic/api/pkg/resources/metrics-server github.com/kubermatic/kubermatic/api/pkg/resources/openvpn github.com/kubermatic/kubermatic/api/pkg/resources/scheduler github.com/kubermatic/kubermatic/api/pkg/resources/usercluster github.com/kubermatic/kubermatic/api/pkg/controller/cluster github.com/kubermatic/kubermatic/api/pkg/controller/container-linux/resources github.com/kubermatic/kubermatic/api/pkg/controller/container-linux github.com/kubermatic/kubermatic/api/pkg/resources/kubestatemetrics github.com/kubermatic/kubermatic/api/pkg/resources/prometheus github.com/kubermatic/kubermatic/api/pkg/docker github.com/kubermatic/kubermatic/api/pkg/validation/nodeupdate github.com/kubermatic/kubermatic/api/cmd/kubeletdnat-controller github.com/kubermatic/kubermatic/api/pkg/serviceaccount github.com/kubermatic/kubermatic/api/pkg/util/errors github.com/kubermatic/kubermatic/api/pkg/handler/auth github.com/kubermatic/kubermatic/api/pkg/handler/v1/common github.com/kubermatic/kubermatic/api/pkg/util/hash github.com/kubermatic/kubermatic/api/pkg/handler/middleware github.com/kubermatic/kubermatic/api/pkg/handler/v1 github.com/kubermatic/kubermatic/api/pkg/handler/v1/addon github.com/kubermatic/kubermatic/api/pkg/defaulting github.com/kubermatic/kubermatic/api/pkg/resources/cluster github.com/kubermatic/kubermatic/api/pkg/resources/machine github.com/kubermatic/kubermatic/api/pkg/handler/v1/cluster github.com/kubermatic/kubermatic/api/pkg/handler/v1/dc github.com/kubermatic/kubermatic/api/pkg/handler/v1/kubernetes-dashboard github.com/kubermatic/kubermatic/api/pkg/machine github.com/kubermatic/kubermatic/api/pkg/handler/v1/node github.com/kubermatic/kubermatic/api/pkg/handler/v1/openshift github.com/kubermatic/kubermatic/api/pkg/handler/v1/presets github.com/kubermatic/kubermatic/api/pkg/handler/v1/project github.com/kubermatic/kubermatic/api/pkg/handler/v1/provider github.com/kubermatic/kubermatic/api/pkg/handler/v1/serviceaccount github.com/kubermatic/kubermatic/api/pkg/handler/v1/ssh github.com/kubermatic/kubermatic/api/pkg/handler/v1/user github.com/kubermatic/kubermatic/api/pkg/metrics github.com/kubermatic/kubermatic/api/pkg/presets github.com/kubermatic/kubermatic/api/pkg/util/informer github.com/kubermatic/kubermatic/api/cmd/kubermatic-api github.com/kubermatic/kubermatic/api/pkg/collectors github.com/kubermatic/kubermatic/api/pkg/controller/clustercomponentdefaulter github.com/kubermatic/kubermatic/api/pkg/controller/update github.com/kubermatic/kubermatic/api/pkg/crd/migrations github.com/kubermatic/kubermatic/api/pkg/leaderelection github.com/kubermatic/kubermatic/api/pkg/metrics/server github.com/kubermatic/kubermatic/api/pkg/crd/operator/v1alpha1 github.com/kubermatic/kubermatic/api/pkg/controller/operator-master/resources/kubermatic github.com/kubermatic/kubermatic/api/cmd/kubermatic-operator github.com/kubermatic/kubermatic/api/pkg/controller/seed-controller-lifecycle github.com/kubermatic/kubermatic/api/pkg/controller/seed-proxy github.com/kubermatic/kubermatic/api/pkg/crd/migrations/master github.com/kubermatic/kubermatic/api/cmd/master-controller-manager github.com/kubermatic/kubermatic/api/cmd/owner-remover github.com/kubermatic/kubermatic/api/pkg/exporters/s3 github.com/kubermatic/kubermatic/api/cmd/s3-exporter github.com/kubermatic/kubermatic/api/cmd/s3-storeuploader github.com/kubermatic/kubermatic/api/pkg/controller/nodecsrapprover github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/clusterautoscaler github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/controller-manager github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/dnat-controller github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/kube-state-metrics github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/kubernetes-dashboard github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/machine-controller github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/metrics-server github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/openshift github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/openvpn github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/prometheus github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/scheduler github.com/kubermatic/kubermatic/api/pkg/controller/usercluster/resources/user-auth github.com/kubermatic/kubermatic/api/pkg/controller/usercluster github.com/kubermatic/kubermatic/api/cmd/user-cluster-controller-manager github.com/kubermatic/kubermatic/api/cmd/userdata-openshift github.com/kubermatic/kubermatic/api/codegen/reconcile github.com/kubermatic/kubermatic/api/pkg/controller/rbac/test github.com/kubermatic/kubermatic/api/pkg/controller/rbac/test/fake github.com/kubermatic/kubermatic/api/pkg/crd/client/clientset/versioned/typed/kubermatic/v1/fake github.com/kubermatic/kubermatic/api/pkg/crd/client/clientset/versioned/fake github.com/kubermatic/kubermatic/api/pkg/handler/test github.com/kubermatic/kubermatic/api/pkg/handler/test/hack github.com/kubermatic/kubermatic/api/pkg/ssh github.com/kubermatic/kubermatic/api/pkg/test github.com/kubermatic/kubermatic/api/pkg/test/e2e/api/utils/oidc-proxy-client github.com/kubermatic/kubermatic/api/cmd/image-loader [github.com/kubermatic/kubermatic/api/cmd/image-loader.test] github.com/kubermatic/kubermatic/api/cmd/kubermatic-controller-manager [github.com/kubermatic/kubermatic/api/cmd/kubermatic-controller-manager.test] github.com/kubermatic/kubermatic/api/cmd/nodeport-proxy/envoy-manager [github.com/kubermatic/kubermatic/api/cmd/nodeport-proxy/envoy-manager.test] github.com/kubermatic/kubermatic/api/cmd/nodeport-proxy/lb-updater [github.com/kubermatic/kubermatic/api/cmd/nodeport-proxy/lb-updater.test] github.com/kubermatic/kubermatic/api/pkg/api/v1_test [github.com/kubermatic/kubermatic/api/pkg/api/v1.test] github.com/kubermatic/kubermatic/api/pkg/clusterdeletion [github.com/kubermatic/kubermatic/api/pkg/clusterdeletion.test] github.com/kubermatic/kubermatic/api/pkg/controller/addon [github.com/kubermatic/kubermatic/api/pkg/controller/addon.test] github.com/kubermatic/kubermatic/api/pkg/controller/addoninstaller [github.com/kubermatic/kubermatic/api/pkg/controller/addoninstaller.test] github.com/kubermatic/kubermatic/api/pkg/controller/backup [github.com/kubermatic/kubermatic/api/pkg/controller/backup.test] github.com/kubermatic/kubermatic/api/pkg/controller/ipam [github.com/kubermatic/kubermatic/api/pkg/controller/ipam.test] github.com/kubermatic/kubermatic/api/pkg/controller/kubeletdnat [github.com/kubermatic/kubermatic/api/pkg/controller/kubeletdnat.test] github.com/kubermatic/kubermatic/api/pkg/controller/monitoring [github.com/kubermatic/kubermatic/api/pkg/controller/monitoring.test] github.com/kubermatic/kubermatic/api/pkg/controller/openshift [github.com/kubermatic/kubermatic/api/pkg/controller/openshift.test] github.com/kubermatic/kubermatic/api/pkg/controller/openshift/resources [github.com/kubermatic/kubermatic/api/pkg/controller/openshift/resources.test] github.com/kubermatic/kubermatic/api/pkg/controller/operator-master [github.com/kubermatic/kubermatic/api/pkg/controller/operator-master.test] github.com/kubermatic/kubermatic/api/pkg/controller/rbac [github.com/kubermatic/kubermatic/api/pkg/controller/rbac.test] github.com/kubermatic/kubermatic/api/pkg/controller/rbac-user-cluster [github.com/kubermatic/kubermatic/api/pkg/controller/rbac-user-cluster.test] github.com/kubermatic/kubermatic/api/pkg/controller/rbac-user-cluster_test [github.com/kubermatic/kubermatic/api/pkg/controller/rbac-user-cluster.test] github.com/kubermatic/kubermatic/api/pkg/controller/seed-sync [github.com/kubermatic/kubermatic/api/pkg/controller/seed-sync.test] github.com/kubermatic/kubermatic/api/pkg/controller/service-account [github.com/kubermatic/kubermatic/api/pkg/controller/service-account.test] github.com/kubermatic/kubermatic/api/pkg/controller/user-cluster-controller-manager/openshift-master-node-labeler [github.com/kubermatic/kubermatic/api/pkg/controller/user-cluster-controller-manager/openshift-master-node-labeler.test] github.com/kubermatic/kubermatic/api/pkg/controller/user-project-binding [github.com/kubermatic/kubermatic/api/pkg/controller/user-project-binding.test] github.com/kubermatic/kubermatic/api/pkg/controller/util [github.com/kubermatic/kubermatic/api/pkg/controller/util.test] github.com/kubermatic/kubermatic/api/pkg/features [github.com/kubermatic/kubermatic/api/pkg/features.test] github.com/kubermatic/kubermatic/api/pkg/handler [github.com/kubermatic/kubermatic/api/pkg/handler.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/addon_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/addon.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/cluster_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/cluster.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/common_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/common.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/dc_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/dc.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/node_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/node.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/presets_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/presets.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/project_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/project.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/provider_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/provider.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/serviceaccount_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/serviceaccount.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/ssh_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/ssh.test] github.com/kubermatic/kubermatic/api/pkg/handler/v1/user_test [github.com/kubermatic/kubermatic/api/pkg/handler/v1/user.test] github.com/kubermatic/kubermatic/api/pkg/kubernetes [github.com/kubermatic/kubermatic/api/pkg/kubernetes.test] github.com/kubermatic/kubermatic/api/pkg/machine_test [github.com/kubermatic/kubermatic/api/pkg/machine.test] github.com/kubermatic/kubermatic/api/pkg/presets_test [github.com/kubermatic/kubermatic/api/pkg/presets.test] github.com/kubermatic/kubermatic/api/pkg/provider [github.com/kubermatic/kubermatic/api/pkg/provider.test] github.com/kubermatic/kubermatic/api/pkg/provider/cloud/aws [github.com/kubermatic/kubermatic/api/pkg/provider/cloud/aws.test] github.com/kubermatic/kubermatic/api/pkg/provider/cloud/vsphere [github.com/kubermatic/kubermatic/api/pkg/provider/cloud/vsphere.test] github.com/kubermatic/kubermatic/api/pkg/provider/kubernetes_test [github.com/kubermatic/kubermatic/api/pkg/provider/kubernetes.test] github.com/kubermatic/kubermatic/api/pkg/resources [github.com/kubermatic/kubermatic/api/pkg/resources.test] github.com/kubermatic/kubermatic/api/pkg/resources/address [github.com/kubermatic/kubermatic/api/pkg/resources/address.test] github.com/kubermatic/kubermatic/api/pkg/resources/apiserver [github.com/kubermatic/kubermatic/api/pkg/resources/apiserver.test] github.com/kubermatic/kubermatic/api/pkg/resources/etcd [github.com/kubermatic/kubermatic/api/pkg/resources/etcd.test] github.com/kubermatic/kubermatic/api/pkg/resources/reconciling [github.com/kubermatic/kubermatic/api/pkg/resources/reconciling.test] github.com/kubermatic/kubermatic/api/pkg/resources/test [github.com/kubermatic/kubermatic/api/pkg/resources/test.test] github.com/kubermatic/kubermatic/api/pkg/semver [github.com/kubermatic/kubermatic/api/pkg/semver.test] github.com/kubermatic/kubermatic/api/pkg/serviceaccount_test [github.com/kubermatic/kubermatic/api/pkg/serviceaccount.test] github.com/kubermatic/kubermatic/api/pkg/storeuploader [github.com/kubermatic/kubermatic/api/pkg/storeuploader.test] github.com/kubermatic/kubermatic/api/pkg/test/e2e/api [github.com/kubermatic/kubermatic/api/pkg/test/e2e/api.test] github.com/kubermatic/kubermatic/api/pkg/userdata/openshift [github.com/kubermatic/kubermatic/api/pkg/userdata/openshift.test] github.com/kubermatic/kubermatic/api/pkg/util/flagopts [github.com/kubermatic/kubermatic/api/pkg/util/flagopts.test] github.com/kubermatic/kubermatic/api/pkg/validation [github.com/kubermatic/kubermatic/api/pkg/validation.test] github.com/kubermatic/kubermatic/api/pkg/validation_test [github.com/kubermatic/kubermatic/api/pkg/validation.test] github.com/kubermatic/kubermatic/api/pkg/validation/seed [github.com/kubermatic/kubermatic/api/pkg/validation/seed.test] github.com/kubermatic/kubermatic/api/pkg/version [github.com/kubermatic/kubermatic/api/pkg/version.test]]"
level=info msg="[runner] worker.1 took 431.342826ms with stages: megacheck: 297.559028ms, govet: 63.204565ms, bodyclose: 22.284407ms, gocyclo: 20.348678ms, misspell: 11.502779ms, golint: 7.856683ms, interfacer: 7.01526ms, errcheck: 620.798µs, gocritic: 463.687µs, gofmt: 222.747µs, ineffassign: 112.183µs, unconvert: 41.895µs, varcheck: 26.147µs, deadcode: 23.793µs, goconst: 11.394µs, structcheck: 8.727µs"
level=info msg="[runner] Workers idle times: "
level=info msg="[runner] Issues before processing: 6, after processing: 0"
level=info msg="[runner] processing took 708.474µs with stages: exclude: 396.115µs, autogenerated_exclude: 124.869µs, identifier_marker: 98.852µs, skip_dirs: 66.535µs, path_prettifier: 9.314µs, cgo: 7.913µs, filename_unadjuster: 1.915µs, nolint: 446ns, diff: 369ns, max_same_issues: 346ns, max_from_linter: 333ns, skip_files: 304ns, exclude-rules: 272ns, uniq_by_line: 247ns, path_shortener: 234ns, source_code: 232ns, max_per_file_from_linter: 178ns"
level=info msg="File cache stats: 2 entries of total size 822B"
level=info msg="Memory: 33 samples, avg is 703.4MB, max is 2349.8MB"
level=info msg="Execution took 10.578883274s"
compilation error during linting
make: *** [Makefile:91: lint] Błąd 1
```

**Comment by floreks** [comment]
`make test` works just fine locally.

/retest

**Comment by floreks** [comment]
Rebased and tests suddenly started to pass.

**Comment by alvaroaleman** [comment]
> Rebased and tests suddenly started to pass.

I do not want to shatter your dreams here, but I think that wasn't the rebase but https://github.com/kubermatic/infra/pull/647 :P

/lgtm

**Pull request merged**